packages/nodes-base/nodes/Google/Gmail/Gmail.node.ts

```diff
@@ -819,17 +819,25 @@ export class Gmail implements INodeType {
 				}
 
 				let executionData = responseData as INodeExecutionData[];
-				if (!['draft', 'message'].includes(resource) && !['get', 'getAll'].includes(operation)) {
+				if (!Array.isArray(executionData)) {
+					executionData = [executionData];
```

Comment:
Instead of doing this check here, we can change line 832 to `this.helpers.returnJsonArray(executionData)` which basically does the same and makes sure we have the `json` key

Reply:
returnJsonArray will wrap data in another json, but result of get and getAll operations has already json key, because we processing binary in those operations